flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/serializer/AvroToProtoSerializer.java

```diff
@@ -297,9 +318,279 @@ private static Object handleLogicalTypeSchema(Schema fieldSchema, Object value)
+                timestamp = (micros ? (Long) value : TimeUnit.MILLISECONDS.toMicros((Long) value));
+            }
+            return validateTimestamp(timestamp);
+        }
+
+        private static Integer validateDate(Integer date) {
+            if (date > 2932896 || date < -719162) {
+                throw new IllegalArgumentException(
+                        String.format(
+                                "Invalid Date '%s' Provided."
+                                        + "\nShould be a Integer value indicating days since Epoch (1970-01-01 00:00:00) "
+                                        + "between %s and %s",
+                                LocalDate.ofEpochDay(date), "0001-01-01", "9999-12-31"));
+            }
+            return date;
+        }
+
+        @VisibleForTesting
+        static Integer convertDate(Object value) {
+            // The value is the number of days since the Unix epoch (1970-01-01).
+            // The valid range is `-719162` (0001-01-01) to `2932896` (9999-12-31).
+            int date;
+            if (value instanceof ReadableInstant) {
+                date = Days.daysBetween(Instant.EPOCH, (ReadableInstant) value).getDays();
+            } else {
+                Preconditions.checkArgument(
+                        value instanceof Integer, "Expecting a value as Integer type (days).");
+                date = (Integer) value;
+            }
+            return validateDate(date);
+        }
+
+        @VisibleForTesting
+        static String convertDateTime(Object value, boolean micros) {
+            if (value instanceof String) {
+                return (String) value;
+            }
+            if (value instanceof Utf8) {
+                return ((Utf8) value).toString();
+            }
+            Preconditions.checkArgument(
+                    value instanceof Long,
+                    String.format(
+                            "Expecting a value as Long type " + "%s.%n Instead %s was obtained",
+                            "Local-Timestamp(micros/millis)", value.getClass()));
+            // Convert to Microseconds if provided in millisecond precision.
+            /* We follow the same steps as that of Timestamp conversion
+            because essentially we have a timestamp - we just need to strip the timezone before insertion.
+             */
+            Long convertedTs = convertTimestamp(value, micros, "Local Timestamp(millis/micros)");
+            Timestamp convertedTimestamp = Timestamp.ofTimeMicroseconds(convertedTs);
+            return LocalDateTime.ofEpochSecond(
+                            convertedTimestamp.getSeconds(),
+                            convertedTimestamp.getNanos(),
+                            ZoneOffset.UTC)
+                    .toString();
+        }
+
+        private static void validateTime(long value) {
+            LocalTime minTime = LocalTime.parse("00:00:00");
+            LocalTime maxTime = LocalTime.parse("23:59:59.999999");
```

Comment:
LocalTime has min and max. Can we use those?

Reply:
Yep, Added.